### PR TITLE
implement category icon service

### LINKED ISSUE #207: (SP: 1) Create category icon service
- [ ] Implement a service for selecting category icons by their names

Image

### PR DESCRIPTION
**Pull request by YaroslavChuiko**
- [x]  Implement a service for selecting category icons by their names

image

### CODE REVIEW

**Review by OlhaOvsiienko — APPROVED** [review]

**Review by YaroslavLys — APPROVED** [review]

**Review by holus-bolus — APPROVED** [review]

File: src/services/category-icon-service.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+type IconKey = keyof typeof CATEGORY_ICONS
+
+function isObjKey<T extends object>(key: PropertyKey, obj: T): key is keyof T {
+  return key in obj
+}
+
+export const getCategoryIcon = (
+  category: IconKey & string
+): typeof CATEGORY_ICONS[IconKey] => {
```
Comment (vlad-khrychov) [inline src/services/category-icon-service.ts]: add import `import { SvgIconComponent } from '@mui/icons-material'`
```suggestion
): SvgIconComponent => {
```

File: src/services/category-icon-service.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  if (isObjKey(category, CATEGORY_ICONS)) {
+    return CATEGORY_ICONS[category]
+  }
+  return CATEGORY_ICONS['LanguageIcon']
+}
```
Comment (vlad-khrychov) [inline src/services/category-icon-service.ts]:
```suggestion
export const getCategoryIcon = (
  category: IconKey
): SvgIconComponent => {
  return CATEGORY_ICONS[category] || CATEGORY_ICONS['LanguageIcon']
}
```

File: tests/unit/services/category-icon-service.spec.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+describe('categoryIconService test', () => {
+  it('should return language icon', () => {
+    iconNames.forEach((iconName) => {
+      const icon = getCategoryIcon(iconName)
+      expect(icon).toEqual(iconName)
+    })
+  })
+
+  it('should return LanguageIcon', () => {
```
Comment (vlad-khrychov) [inline tests/unit/services/category-icon-service.spec.js]:
```suggestion
  it('should return LanguageIcon for invalid icon name', () => {
```

File: src/services/category-icon-service.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  MusicNoteIcon: MusicNoteIcon,
+  ScienceIcon: ScienceIcon,
+  LegendToggleIcon: LegendToggleIcon
+} as const
+
+type IconKey = keyof typeof CATEGORY_ICONS
+
+function isObjKey<T extends object>(key: PropertyKey, obj: T): key is keyof T {
+  return key in obj
+}
```
Comment (vlad-khrychov) [inline src/services/category-icon-service.ts]: I do not think that we need this function.

**Review by vlad-khrychov — CHANGES_REQUESTED** [review]

**Review by vlad-khrychov — APPROVED** [review]

**Pull request merged**